packages/matter.js/src/behavior/definitions/switch/SwitchServer.ts

```diff
@@ -4,14 +4,280 @@
- * {@link SwitchServer.with} to specialize the class for the features your implementation supports.
+ * This implementation includes all features of {@link WindowCovering.Cluster} and implements all mandatory commands.
+ * You should use {@link WindowCoveringServer.with} to specialize the class for the features your implementation
+ * supports.
+ *
+ * To support all features and events the implementation adds some custom state attributes that can - or are required
+ * to - be used. These are:
+ * * `rawPosition` - The raw position of the switch. This is the position as reported by the device and is not yet
+ *  debounced. If the position is already debounced, just set the `currentPosition` attribute. The debouncing uses the
+ *  `debounceDelay` attribute value as timeframe in milliseconds and sets the `currentPosition` attribute when the value
+ *  is stable for this timeframe.
+ * * `debounceDelay` - The timeframe in milliseconds to wait until a newly reported position is considered stable.
+ * * `longPressDelay` - The timeframe in milliseconds to wait until a position is considered "long" pressed. This
+ *  attribute is REQUIRED if the device supports the Momentary Switch LongPress (MSL) feature.
+ * * `multiPressDelay` - The timeframe in milliseconds starting with a stable release to detect multi-presses. This
+ *  attribute is REQUIRED if the device supports the Momentary Switch MultiPress (MSM) feature.
+ * * `momentaryNeutralPosition` - The number of the position considered as the neutral position for the momentary switch.
+ *  This defaults to position 0 but can be changed by settings this attribute. it is available as soon as the Momentary
+ *  Switch (MS) feature is used.
  */
-export class SwitchServer extends SwitchBehavior {}
+export class SwitchServerLogic extends SwitchServerBase {
+    protected declare internal: SwitchServerLogic.Internal;
+    declare state: SwitchServerLogic.State;
+    declare events: SwitchServerLogic.Events;
+
+    // Enhance Schema to define conformance for some of the additional state attributes
+    static {
+        SwitchServerLogic.schema?.children.push(
+            FieldElement({ name: "longPressDelay", type: "uint32", quality: "M", conformance: "MSL" }),
+            FieldElement({ name: "multiPressDelay", type: "uint32", quality: "M", conformance: "MSM" }),
+            FieldElement({ name: "momentaryNeutralPosition", type: "uint8", quality: "O", conformance: "MS" }),
+        );
+    }
+
+    override initialize() {
+        this.state.rawPosition = this.state.currentPosition;
+        this.internal.currentUnstablePosition = this.state.currentPosition;
+        this.internal.previouslyReportedPosition = this.state.currentPosition;
+
+        // Validate positions set to respect constraints, TODO: Remove with 1.3 update
+        this.reactTo(this.events.rawPosition$Changing, this.#assertPositionInRange, { offline: true });
```

Comment:
I didn't see any tests that cover these, did you test manually?  The "offline" flag I think would make it run out-of-band so any errors these throw would be logged but wouldn't prevent the write.

See `ReactorBacking.#react` in Reactors.ts...  If "offline" it ignores the parent context.  Then when there's no parent context it captures any errors and passes them to `#unhandledError` which logs them.

Reply:
I add tests. And yes I can remove offline Here I think.